plugins/pulp_docker/plugins/importers/importer.py

```diff
@@ -310,9 +316,49 @@ def _import_manifest(conduit, unit, dest_repo):
         for blob in models.Blob.objects.filter(digest__in=sorted(blob_digests)):
             units_added |= set(DockerImporter._import_blob(conduit, blob, dest_repo))
 
+        # Add manifests
+        repository.associate_single_unit(dest_repo, unit)
+        units_added.add(unit)
+
         return units_added
 
     @staticmethod
+    def _import_manifest_list(conduit, unit, dest_repo):
+        """
+        Import a Manifest List and its referenced image manifests.
+
+        :param conduit:   provides access to relevant Pulp functionality
+        :type  conduit:   pulp.plugins.conduits.unit_import.ImportUnitConduit
+        :param unit:      The Manifest List to import
+        :type  unit:      pulp_docker.plugins.Model.ManifestList
+        :param dest_repo: The destination repository that the ManifestList is being imported to.
+        :type  dest_repo: pulp.server.db.model.Repository
+        :return:          list of Units that were copied to the destination repository
+        :rtype:           list
+        """
+
+        units_added = set()
+
+        # Catalog referenced manifests
+        manifest_digests = set()
+        for manifest in unit.manifests:
+            manifest_digests.add(manifest)
+
+        if unit.amd64_digest:
+            manifest_digests.add(unit.amd64_digest)
```

Comment:
it would not, in case in the unit.manifests there was schema2, that means in amd64_digest would be the converted manifest to schema1. in case in unit.manifests there already was schema1, since mainfests_digests is a set, it would not contain duplicates.